include/fmt/format.h

```diff
@@ -1798,22 +1818,76 @@ auto write_float(OutputIt out, const DecimalFP& fp,
   });
 }
 
+#ifdef __cpp_lib_bit_cast
+template <typename T, FMT_ENABLE_IF(is_fast_float<T>::value)>
+constexpr bool is_infinite_fast_float(T value) {
+  using floaty =
+      std::conditional_t<std::is_same<T, long double>::value, double, T>;
+  using uint = typename dragonbox::float_info<floaty>::carrier_uint;
+  constexpr auto significand_bits =
+      dragonbox::float_info<floaty>::significand_bits;
+  auto bits = std::bit_cast<uint>(value);
+  return (bits & exponent_mask<floaty>()) &&
+         !(bits & ((uint(1) << significand_bits) - 1));
+}
+
+template <typename T, FMT_ENABLE_IF(is_fast_float<T>::value)>
+constexpr bool is_finite_fast_float(T value) {
+  using floaty =
+      std::conditional_t<std::is_same<T, long double>::value, double, T>;
+  using uint = typename dragonbox::float_info<floaty>::carrier_uint;
+  auto bits = std::bit_cast<uint>(value);
+  return (bits & exponent_mask<floaty>()) != exponent_mask<floaty>();
+}
+#endif
+
 template <typename Char, typename OutputIt, typename T,
           FMT_ENABLE_IF(std::is_floating_point<T>::value)>
-auto write(OutputIt out, T value, basic_format_specs<Char> specs,
-           locale_ref loc = {}) -> OutputIt {
+FMT_CONSTEXPR20 auto write(OutputIt out, T value,
+                           basic_format_specs<Char> specs, locale_ref loc = {})
+    -> OutputIt {
   if (const_check(!is_supported_floating_point(value))) return out;
   float_specs fspecs = parse_float_type_spec(specs);
   fspecs.sign = specs.sign;
-  if (std::signbit(value)) {  // value < 0 is false for NaN so use signbit.
+  bool is_negative;
+#ifdef __cpp_if_constexpr
+  if (is_constant_evaluated()) {
+    // bit check below is valid only for "fast float" types
+    if constexpr (is_fast_float<T>::value) {
+      using floaty =
+          conditional_t<std::is_same<T, long double>::value, double, T>;
+      using uint = typename dragonbox::float_info<floaty>::carrier_uint;
+      auto bits = bit_cast<uint>(value);
+      is_negative = (bits & (uint(1) << (num_bits<uint>() - 1))) != 0;
+    } else {
+      FMT_ASSERT(false, "floating point type is not supported");
+    }
+  } else
+#endif
```

Comment:
`is_constant_evaluated` is available even if `__cpp_if_constexpr` is not defined so please move the `#ifdef __cpp_if_constexpr` block into the if branch to make the formatting less weird.

Reply:
Done